### PR TITLE
GSREN3D-487: Solar panel placement with matrix of grid

### PR DESCRIPTION
**Pull request by ismailsunni**
### JIRA issue

https://jira.camptocamp.com/browse/GSREN3D-487

### Description

- [x] Fix the matrix format
- [x] Remove the old algorithm
- ~Integrate with the main workflow~ -> will be done on another PR (continuing Yohann's PR)

### Screenshots

Sample result:
- very dark red: not usable roof grid
- green: usable roof grid but not covered by solar panel
- other colors: a grid covered by a solar panel (2x4 grid with the same color)

image

### CODE REVIEW

File: src/algorithm/solarPanelPlacement.ts [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+) {
+  if (debug) {
+    fs.writeFileSync(
+      path.join(__dirname, outputFileName),
+      JSON.stringify(feature)
+    )
+  }
+}
+
+function createSolarPanel(
```
Comment (toni-garcia-camptocamp) [inline src/algorithm/solarPanelPlacement.ts:42]: Can you add one example with the value of index and horizontal and the result of function please

File: src/algorithm/solarPanelPlacement.ts [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+    }
+  }
+
+  const solarPanelGrid: SolarPanelGrid = {
+    gridIndexes: indexes,
+  }
+  return solarPanelGrid
+}
+
+function isAllSolarPanelGridUsable(
```
Comment (toni-garcia-camptocamp) [inline src/algorithm/solarPanelPlacement.ts:63]: Can you add comment for explain this function please

File: src/algorithm/solarPanelPlacement.ts [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+      ].usable
+    ) {
+      return false
+    }
+  }
+
+  return true
+}
+
+function isAllSolarPanelGridNotCovered(
```
Comment (toni-garcia-camptocamp) [inline src/algorithm/solarPanelPlacement.ts:87]: Same here, comment please :)
With one example if possible

File: src/algorithm/solarPanelPlacement.ts [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+      coveredGrids[gridIndex[1] * matrix.length + gridIndex[0]] = i
+    })
+  }
+
+  for (let i = 0; i < matrix.length; i++) {
+    for (let j = 0; j < matrix[0].length; j++) {
+      const x: number = i * -length
+      const y: number = j * length
+      const solarPanelIndex = coveredGrids[j * matrix.length + i]
+      const grid: Feature<Polygon, Properties> = polygon(
```
Comment (toni-garcia-camptocamp) [inline src/algorithm/solarPanelPlacement.ts:199]: Can you explain here in comment please, It's a pretty complicated part to understand.

File: src/algorithm/__tests__/solarPanelPlacement.spec.ts [hunk 5]
```diff
@@ -0,0 +1,57 @@
+import { describe, expect, test } from 'vitest'
```
Comment (toni-garcia-camptocamp) [inline src/algorithm/__tests__/solarPanelPlacement.spec.ts:1]: Maybe put the tests in the folder : src/test/algorithm with all the other test

**Review by toni-garcia-camptocamp — COMMENTED** [review]
Very nice work ! 
Your code work fine, just add some comments and it's good :)

File: src/algorithm/solarPanelPlacement.ts [hunk 6]
```diff
@@ -0,0 +1,259 @@
+import {
```
[the remark this answers is not in the document]
Comment (ismailsunni) [reply src/algorithm/solarPanelPlacement.ts:1]: I think so far it's fast, we can check again later if needed

**Pull request merged**